libfreerdp/core/update.c

```diff
@@ -2122,7 +2122,7 @@ BOOL update_read_refresh_rect(rdpUpdate* update, wStream* s)
 	Stream_Read_UINT8(s, numberOfAreas);
 	Stream_Seek(s, 3); /* pad3Octects */
 
-	if (Stream_GetRemainingLength(s) < ((size_t)numberOfAreas * 4 * 2))
+	if (Stream_GetRemainingLength(s) / 2 / 4 < numberOfAreas)
 		return FALSE;
```

Comment:
`/ 8` would be more explicit, I had to re-read it twice to be sure it was correct

Reply:
kept it the same as it was before (e.g. 4 coordinates, 2 bytes size for each)

Reply:
done.